### PR TITLE
refactor: Convert Utils to Kotlin 

### PR DESCRIPTION
**Pull request by david-allison**
This one was horrible, lots of work + incorrect automatic conversions. Please review carefully.

Closes #11920

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Utils.kt [hunk 1]
```diff
@@ -847,139 +869,142 @@ public class Utils {
...
+            ankiDroidFonts = context.assets.list("fonts")
+        } catch (e: IOException) {
+            Timber.e(e, "Error on retrieving ankidroid fonts")
         }
-        List<AnkiFont> fonts = new ArrayList<>(fontsCount);
-        for (int i = 0; i < fontsCount; i++) {
-            String filePath = fontsList[i].getAbsolutePath();
-            String filePathExtension = splitFilename(filePath)[1];
-            for (String fontExtension : FONT_FILE_EXTENSIONS) {
+        val fonts: MutableList<AnkiFont> = ArrayList(fontsCount)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Utils.kt:958]: @KotlinCleanup("See if code in for loop an be improved")

**Review by lukstbit — CHANGES_REQUESTED** [review]

**Comment by lukstbit** [comment]
@david-allison Got a conflict...

**Review by lukstbit — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
Another massive one, amazing
pending local CI re-run as these Kotlin ones are in my merge train

**Comment by mikehardy** [comment]
@david-allison good to go except merge-train local CI run just caught this one from Models cross-contamination:

```

> Task :AnkiDroid:compilePlayDebugKotlin FAILED
e: /home/mike/work/AnkiDroid/Anki-Android-Upstream/AnkiDroid/src/main/java/com/ichi2/libanki/Models.kt: (470, 46): Type mismatch: inferred type is Array<String?> but Array<String> was expected
e: /home/mike/work/AnkiDroid/Anki-Android-Upstream/AnkiDroid/src/main/java/com/ichi2/libanki/Models.kt: (470, 59): Type mismatch: inferred type is Array<String> but Array<String?> was expected
e: /home/mike/work/AnkiDroid/Anki-Android-Upstream/AnkiDroid/src/main/java/com/ichi2/libanki/Models.kt: (626, 43): Type mismatch: inferred type is String? but String was expected

> Task :AnkiDroid:compilePlayReleaseKotlin FAILED
e: /home/mike/work/AnkiDroid/Anki-Android-Upstream/AnkiDroid/src/main/java/com/ichi2/libanki/Models.kt: (470, 46): Type mismatch: inferred type is Array<String?> but Array<String> was expected
e: /home/mike/work/AnkiDroid/Anki-Android-Upstream/AnkiDroid/src/main/java/com/ichi2/libanki/Models.kt: (470, 59): Type mismatch: inferred type is Array<String> but Array<String?> was expected
e: /home/mike/work/AnkiDroid/Anki-Android-Upstream/AnkiDroid/src/main/java/com/ichi2/libanki/Models.kt: (626, 43): Type mismatch: inferred type is String? but String was expected

```

**Comment by mikehardy** [comment]
This one has nullability cross-contamination from the Utils convertion:

> e: /home/mike/work/AnkiDroid/Anki-Android-Upstream/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt: (526, 67): Type mismatch: inferred type is Array<String?> but Array<String> was expected

I had a look at fixing it but the variable is defined as an array of nulls to start, not sure how far we want to go on forcing that or where the best place is. Could be as simple as a cast since it's a test and if it blows up in testing, great, but I didn't feel confident

**Comment by david-allison** [comment]
Re-pushed

**Comment by mikehardy** [comment]
Clean!

**Pull request merged**